packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts

```diff
@@ -196,6 +323,106 @@ export abstract class VpcV2Base extends Resource implements IVpcV2 {
     });
   }
 
+  /**
+   * Adds a new Egress Only Internet Gateway to this VPC and defines a new route
+   * to the route table of given subnets
+   * @default in case of no input subnets, no route is created.
+   */
+  public addEgressOnlyInternetGateway(options?: EgressOnlyInternetGatewayOptions): void {
+    const egw = new EgressOnlyInternetGateway(this, 'EgressOnlyGW', {
+      vpc: this,
+    });
+
+    const useIpv6 = (this.secondaryCidrBlock.some((secondaryAddress) => secondaryAddress.amazonProvidedIpv6CidrBlock === true ||
+    secondaryAddress.ipv6IpamPoolId != undefined))? true : false;
```

Comment:
What is this doing?

Reply:
Since we can have multiple secondary address attached to VPC, this is to check if any of the secondary address is a IPv6 meaning created using `IPAMIpv6` or is a `AmazonProvidedIpv6`